### PR TITLE
descheduler v0.31: update docs and manifests

### PR DESCRIPTION
**Pull request by a7i**
Closes https://github.com/kubernetes-sigs/descheduler/issues/1492

Similar to https://github.com/kubernetes-sigs/descheduler/pull/617/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ that version's release branch, as listed below:
 
 |Descheduler Version|Docs link|
 |---|---|
+|v0.31.x|[`release-1.31`](https://github.com/kubernetes-sigs/descheduler/blob/release-1.31/README.md)|
 |v0.30.x|[`release-1.30`](https://github.com/kubernetes-sigs/descheduler/blob/release-1.30/README.md)|
 |v0.29.x|[`release-1.29`](https://github.com/kubernetes-sigs/descheduler/blob/release-1.29/README.md)|
 |v0.28.x|[`release-1.28`](https://github.com/kubernetes-sigs/descheduler/blob/release-1.28/README.md)|
```
Comment (ingvagabund) [inline README.md:44]: We could probably truncate some of the oldest releases. Keeping 1.28+?

File: README.md [hunk 2]
```diff
@@ -38,6 +38,7 @@ that version's release branch, as listed below:
```
Comment (ingvagabund) [inline README.md:38]: such as `registry.k8s.io/descheduler/descheduler:v0.26.1` -> such as `registry.k8s.io/descheduler/descheduler:v0.31.0`

File: docs/user-guide.md [hunk 3]
```diff
@@ -4,6 +4,7 @@ Starting with descheduler release v0.10.0 container images are available in the
 
 Descheduler Version | Container Image                                 | Architectures           |
 ------------------- |-------------------------------------------------|-------------------------|
+v0.31.0             | registry.k8s.io/descheduler/descheduler:v0.31.0 | AMD64<br>ARM64<br>ARMv7 |
 v0.30.1             | registry.k8s.io/descheduler/descheduler:v0.30.1 | AMD64<br>ARM64<br>ARMv7 |
 v0.30.0             | registry.k8s.io/descheduler/descheduler:v0.30.0 | AMD64<br>ARM64<br>ARMv7 |
 v0.29.0             | registry.k8s.io/descheduler/descheduler:v0.29.0 | AMD64<br>ARM64<br>ARMv7 |
```
Comment (ingvagabund) [inline docs/user-guide.md:10]: Should we truncate the older releases and keeping only the supported? E.g. 0.28.0+?

File: kubernetes/cronjob/cronjob.yaml [hunk 4]
```diff
@@ -16,7 +16,7 @@ spec:
           priorityClassName: system-cluster-critical
           containers:
           - name: descheduler
-            image: registry.k8s.io/descheduler/descheduler:v0.30.1
+            image: registry.k8s.io/descheduler/descheduler:v0.31.0
```
[the remark this answers is not in the document]
Comment (ingvagabund) [reply kubernetes/cronjob/cronjob.yaml:19]: Why not?

[the remark this answers is not in the document]
Comment (a7i) [reply kubernetes/cronjob/cronjob.yaml:19]: that's right

```
Run kubectl create -f kubernetes/base/rbac.yaml
  kubectl create -f kubernetes/base/rbac.yaml
  kubectl create -f test/manifests/v1alpha[2](https://github.com/kubernetes-sigs/descheduler/actions/runs/10614235018/job/29419704631?pr=1506#step:7:2)/configmap.yaml
  kubectl create -f kubernetes/deployment/deployment.yaml
  shell: /usr/bin/bash -e {0}
clusterrole.rbac.authorization.k8s.io/descheduler-cluster-role created
serviceaccount/descheduler-sa created
clusterrolebinding.rbac.authorization.k8s.io/descheduler-cluster-role-binding created
configmap/descheduler-policy-configmap created
deployment.apps/descheduler created
```

```
Run kubectl wait --for=condition=Available --timeout=60s deployment descheduler -n kube-system
  kubectl wait --for=condition=Available --timeout=60s deployment descheduler -n kube-system
  shell: /usr/bin/bash -e {0}
error: timed out waiting for the condition on deployments/descheduler
Error: Process completed with exit code 1.
```

**Pull request merged**